### PR TITLE
Make AppVeyor tests pass

### PR DESCRIPTION
**Pull request by rjmholt**
The `FindNoReferencesOnEmptyLine` test is hanging in AppVeyor (not on a local Windows machine).

This skips it.

We should work out why this is though, especially since it isn't even an `async` test...

I've also re-enabled a bunch of previously skipped tests, because they seem to be passing.

EDIT: Turns out I had to add a timeout to the task tests (even though none of them actually time out...)

### CODE REVIEW

File: test/PowerShellEditorServices.Test/Language/LanguageServiceTests.cs [hunk 1]
```diff
@@ -415,4 +415,17 @@ private FindOccurrencesResult FindSymbolsInFile(ScriptRegion scriptRegion)
                     GetScriptFile(scriptRegion));
         }
     }
+
+    internal static class TaskExtensions
+    {
+        public static async Task<T> RunWithTimeout<T>(this Task<T> task, int timeoutMillis = 10000)
+        {
+            if (await Task.WhenAny(task, Task.Delay(timeoutMillis)) == task)
```
Comment (SeeminglyScience) [inline test/PowerShellEditorServices.Test/Language/LanguageServiceTests.cs]: Might be a little easier to use the tasks' `AsyncWaitHandle`

```csharp
if (!((IAsyncResult)task).AsyncWaitHandle.WaitOne(timeoutMillis))
{
    throw new TimeoutException();
}

return await task;
```

Don't know which route is better, just giving an alternate

Comment (rjmholt) [reply test/PowerShellEditorServices.Test/Language/LanguageServiceTests.cs]: Ah! That definitely feels more "as designed" than my abuse of tasks

File: test/PowerShellEditorServices.Test.Host/LanguageServerTests.cs [hunk 2]
```diff
@@ -90,7 +90,7 @@ public async Task ServiceReturnsSyntaxErrors()
                 string.IsNullOrEmpty(diagnostics.Diagnostics[0].Message));
         }
 
-        [Fact(Skip = "Skipping until Script Analyzer integration is added back")]
```
Comment (tylerl0706) [inline test/PowerShellEditorServices.Test.Host/LanguageServerTests.cs]: lol

File: test/PowerShellEditorServices.Test.Host/LanguageServerTests.cs [hunk 3]
```diff
@@ -292,7 +292,7 @@ public async Task FindsReferencesOfVariable()
             Assert.Equal(12, locations[2].Range.Start.Character);
         }
 
-        [Fact]
+        [Fact(Skip = "AppVeyor is currently hanging on this test it seems")]
```
Comment (tylerl0706) [inline test/PowerShellEditorServices.Test.Host/LanguageServerTests.cs]: is this test the culprit?

File: test/PowerShellEditorServices.Test.Host/LanguageServerTests.cs (hunk 2 above)
Comment (rjmholt) [reply test/PowerShellEditorServices.Test.Host/LanguageServerTests.cs]: Yeah

File: test/PowerShellEditorServices.Test.Host/LanguageServerTests.cs (hunk 3 above)
Comment (rjmholt) [reply test/PowerShellEditorServices.Test.Host/LanguageServerTests.cs]: It *seems* to be. It's strange. *Just* skipping this test doesn't work, and *just* putting timeouts on the `async` tests doesn't work. But both together passes, consistently. Naturally, "if you fix it but you don't know how, you didn't fix it"...

For now I just want to get tests running and passing in CI, but I think this is another signal that reworking the tests is needed

Comment (SeeminglyScience) [reply test/PowerShellEditorServices.Test.Host/LanguageServerTests.cs]: I wonder if the timeout isn't doing anything for some tests because there isn't an `await` soon enough. From what I understand async methods are ran on the same thread they are started until they hit the first `await` keyword.  There may be something that *appears* to be async but doesn't actually use the keyword enough.

If you want you could try forcing the test into another thread by running it with `Task.Run`

Comment (rjmholt) [reply test/PowerShellEditorServices.Test.Host/LanguageServerTests.cs]: It's back to hanging again :angry:

Comment (SeeminglyScience) [reply test/PowerShellEditorServices.Test.Host/LanguageServerTests.cs]: The initial invocation of the task needs to be in the `Task.Run` block.  My guess is that the method never actually runs a task in the first place, it just hangs before it hits an `await`.

If you wanted to make it a utility method you could do something like

```csharp
async Task<T> RunWithTimeout<T>(Func<Task<T>> test, int timeout)
{
    ((IAsyncResult)Task.Run(test)).WaitOne(..etc
```

Comment (rjmholt) [reply test/PowerShellEditorServices.Test.Host/LanguageServerTests.cs]: Bugger, still hanging

Comment (rjmholt) [reply test/PowerShellEditorServices.Test.Host/LanguageServerTests.cs]: Ok, reverting back to the last commit that passed this passes again... :unamused:

Comment (rjmholt) [reply test/PowerShellEditorServices.Test.Host/LanguageServerTests.cs]: I'm just going to merge this for now so the other PRs can test properly

Comment (rkeithhill) [reply test/PowerShellEditorServices.Test.Host/LanguageServerTests.cs]: >  async methods are ran on the same thread they are started until they hit the first await keyword.

Even then, IIRC an async operation can complete during the initial awaited async call if the operation determines it can return the result right away.

**Comment by rjmholt** [comment]
I've reverted this PR since it didn't actually make the tests pass.

At this point I think trying to fix the hang is going to be a bad investment of effort. I'm already working on tests, so I may as well just do my best to rewrite these tests in a way that makes them work.
